core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
```

Comment:
Since fCanLoadClassInfo is atomic, we should probably short-circuit on it rather than `fClassInfo`. That is, we should use `if (!fCanLoadClassInfo || fClassInfo) return;` then set `fCanLoadClassInfo` to false once it's already loaded.